lua/luasnip/config.lua

```diff
@@ -34,7 +34,16 @@ end
 -- declare here to use in set_config.
 local c
 
+---@class UserConfig
+---@field ext_opts table<string, table>
+---@field history boolean
+---@field update_events string[]
+---@field updateevents string[]
+
+---@alias SetConfig fun(user_config: UserConfig)
+
 c = {
+	---@type SetConfig
```

Comment:
Is the SetConfig-alias necessary? `---@type fun ...` seems more direct :D

Reply:
I'm grateful you find the time to watch my contribution :) Don't hesitate if you wanna clear things up vocally, I am reachable 👍

Usually, I use alias when I need to use the same type twice. Isn't ConfigSetup used somewhere else?

Reply:
Ah, maybe for `c.setup`? But that seems like something luals should be able to resolve automatically

> Don't hesitate if you wanna clear things up vocally, I am reachable 👍

Cool, I'll keep it in mind :+1:

Reply:
(e.g. quick search didn't show any other usage :P)

Reply:
Sorry, I am on my phone and it's a bit difficult to search right now, but if it's alone, it's possible I deleted it and then didn't update the alias.

Reply:
Ahh okay :+1: